be/src/exec/pipeline/hashjoin/spillable_hash_join_build_operator.cpp

```diff
@@ -153,9 +166,14 @@ Status SpillableHashJoinBuildOperator::push_chunk(RuntimeState* state, const Chu
         return Status::OK();
     }
 
-    // TODO: materialize chunk (const/nullable)
-
     auto& ht = _join_builder->hash_join_builder()->hash_table();
+    // Estimate the appropriate number of partitions
+    if (_is_first_time_spill && ht.get_row_count() > 0) {
+        // We estimate the size of the hash table to be twice the size of the already input hash table
+        auto num_partitions = ht.mem_usage() * 2 / _join_builder->spiller()->options().spill_mem_table_bytes_size;
```

Comment:
how to ensure num_partitions is greater than 0?

Reply:
see detail implements in reset_partitions
```
Status PartitionedSpillerWriter::reset_partition(RuntimeState* state, size_t num_partitions) {
    num_partitions = BitUtil::next_power_of_two(num_partitions);
    num_partitions = std::min<size_t>(num_partitions, 1 << max_partition_level);
    num_partitions = std::max<size_t>(num_partitions, _spiller->options().init_partition_nums);
    ......
}
```